flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistry.java

```diff
@@ -198,6 +204,14 @@ public boolean isShutdown() {
 	 * Shuts down this registry and the associated {@link MetricReporter}.
 	 */
 	public void shutdown() {
+		Future<Boolean> stopFuture = null;
```

Comment:
why do you split the additions in 2 blocks? Could you not merge them with the sole condition that the MQS is not null?